helsingborg-stad/app-mitt-helsingborg
# Feat/list completions in detail view

## Explain the changes you’ve made
Be able to show completions in the CaseSummary screen. Also added logic for doing text replacement within a case item.

## Explain why these changes are made
This is a task within the feature ow showing completions to the user.

## Explain your solution
When the case items are fetched, we immediately run the text replacement function so that we are certain that text replacement has occurred. 

The new text replacement function changes text within a case based on a set of rules which specifies which key (#KEY) in a case that should be replaced. The rules also specifies which path it should take the value from and a path to where the property with the key is.

## How to test
1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios`
3. Make sure you have a case with the status type of `active:completionRequired:viva`
4. Make sure that the current form have encryption type of `decrypted`
5. Make sure that the case item has `periodstartdate` and `completionduedate` in the `application` property in workflow

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots
![simulator_screenshot_29E9F357-1818-49C2-A3E9-9C5AF1A646E0](https://user-images.githubusercontent.com/81250970/148893964-3a4656bb-68c8-49ef-937a-b8bba51c9ae7.png)
